### ISSUE TITLE
Errors - silently fails upon extracting

### ISSUE THREAD

**Issue by gooor** [ZJONSSON/node-unzipper#98]
It doesn't throw errors, works unexpectedly, better to just install zip and do child_process. Come on, I just wasted 3 hours of my life trying to use it and it's complete mess

**Comment by ZJONSSON** [comment]
Sorry that you wasted 3 hours of your life.  This is an open source project where anyone can contribute fixes and solutions to make it better.    If you wanted to be constructive you could have included detail of what you had problems with and suggestions for fixes.    It might be related to https://github.com/ZJONSSON/node-unzipper/pull/92 which will land soon.

**Comment by gooor** [comment]
Okay excuse me  for my tone - it was wrong!

Anyway I think if it's known issue it should be stated in README that this might cause problems while extracting full zip.

```javascript
  const zipper = fs.createReadStream(filePath).pipe(unzipper.Extract({ path: path.resolve(path.dirname(require.main.filename), 'dist') }));
  zipper.on('close', () => {
    console.log('done');
  });
  zipper.on('error', (error) => {
    console.log(error);
  });
```
1. zip is not fully extracted (sometimes it's only 2 files sometimes its 3 or 4 while there are 10 files inside
2. neither close or error is called so it's looks like #92 
3. Most common usage of it would be to unzip file uploaded from browser - if there's something wrong with it (ie. file corrupted) then it will just silently fail and there's no way of knowing that
But if so I cannot rely on it at all :(

**Comment by ZJONSSON** [comment]
Thank you for this report.   I just extracted a bunch of zip files and was unable to recreate this issue (i.e. the `extract` matches the content of manual uncompress).  This is also confirmed by [unit tests](https://github.com/ZJONSSON/node-unzipper/tree/master/test) that are [tested on multiple node versions](https://travis-ci.org/ZJONSSON/node-unzipper) before every release.92 

Please note that the fix to #92 was released yesterday and the most current version of `unzipper` is `0.9.8`

To help me debug your particular situation, please upload a failing zip file (or send to me at ziggy.jonsson.nyc@gmail.com) and/or give me more information about your setup, i.e. OS, node version and unzipper version.

**Comment by ZJONSSON** [comment]
@gooor - following up on ^^^

**Comment by ZJONSSON** [comment]
Thanks @gooor.   Obviously I can only be helpful if you share the full detail (sample file, your os etc).   

Please note that in your example you do not explicitly need to write the file to fs unless you need to retain the zip file.   You should be able to:

```js
http.get(url, async response => async new Promise( (resolve, reject) => {
  response.pipe(unzipper.Extract({path})
    .on('close', resolve)
    .on('error', reject);
}));
```

and furthermore, if you are only interested in particular files inside the zip (not all of them), you can download/unzip only the files of interest from inside the zipfile (not the other files) directly (see README and also [webunzipper](https://www.npmjs.com/package/webunzip) (a cli tool based on unzipper).  Here is an example of fetching smaller files out of 500mb within a zip file:

![webunzipper](https://user-images.githubusercontent.com/1082488/51766667-242cfa00-20a9-11e9-86a9-a653206761e6.gif)


Nevertheless, if you ware having issues with extract it's important to get to the bottom of this, so I appreciate you help

**Comment by ZJONSSON** [comment]
Thanks @goor that is very helpful.  Will take a look!

**Comment by willpox** [comment]
I am also experiencing this problem.

I am missing several files using unzipper. However, if I unzip the zip file manually, all the files are in there.

**Comment by mmajcica** [comment]
I'm experiencing a similar problem. In case of the mixed slashes in my destination path, (e.g. on win "C:\temp/anotherDir") the extract method silently fails. A simple path.normalize over the path makes extract succeed.

**Comment by ZJONSSON** [comment]
Thanks @mmajcica,  can you submit a PR?

**Comment by dangear** [comment]
@ZJONSSON Hi! I have the same error when use it with node ftp client.

```
const fs = require('fs');
const path = require('path');
const unzipper = require('unzipper');
const async = require('async');
const Client = require('ftp');

var c = new Client(); //init ftp client
var connectionProperties = { 
    host: "PM Please if you need this for test",
    user: "PM Please if you need this for test",
    password: "PM Please if you need this for test"
};

var year='2019';
var folders=['purchaseNotice','purchaseNoticeAE','purchaseNoticeAE94','purchaseNoticeEP','purchaseNoticeOA','purchaseNoticeOK','purchaseNoticeZK'];

var listFile = function(ftp_files_folder, region){ //Get folder file list
  return new Promise(function(resolve, reject){
    c.list(ftp_files_folder,function (err, list) {
      if (err) reject(err)
      var mask=region+'_'+year; //mask for search
      var paths=list.filter(path=>{return path.name.split(mask).length>1}).map(path=>{return path.name});
      resolve(paths) //send files list
    })
  })
}

var listRegions = function(ftp_main_folder){ //Get ftp subfolders
  return new Promise(function(resolve, reject){
    c.list(ftp_main_folder,function (err, list) {
      if (err) reject(err)
      var paths=list.filter(path=>{
        if(path.type=='d' && path.name!='archive') return path;
      }).map(path=>{return path.name});
      resolve(paths) //send files list
    })
  })
}

c.connect(connectionProperties); //connect ftp client

listRegions('/out/published/').then((all_regions)=>{ //get folder for all regions from ftp
  async.mapLimit(all_regions, 1, function(region, callback){ //get every region and check for exist files in folders from folders[array]
    console.log('\x1b[35m','[1]: Get folders for '+region,'\x1b[0m');
    async.mapLimit(folders, 1, function(folder, callback){ //get every folder from folder[array]
      var ftp_region_folders='/out/published/'+region+'/'+folder+'/daily';
      listFile(ftp_region_folders, region).then(ftp_region_folder_filelist=>{ //get file list from ftp folder like folders[array]
        console.log('\x1b[36m','[2]: Get files form '+region+'/'+folder+' that exist: '+ftp_region_folder_filelist.length+' files','\x1b[0m');
        async.mapLimit(ftp_region_folder_filelist, 1, function(file, callback){ //get every file from file list
          var ftp_region_folder_filepath=ftp_region_folders+'/'+file; //ftp file path
          var local_filepath=__dirname+'/alldata/'+region+'/'+folder+'/'+file.split('.xml').shift(); //local folder for save
          if(!fs.existsSync(local_filepath)) { //if subfolder not exist
            console.log('[3.1]: Try to download: '+ftp_region_folder_filepath);
            c.get(ftp_region_folder_filepath, function (err, stream) { //get file from ftp file path
              if(err) callback(err);			    
              stream.pipe(unzipper.Extract({path: path.normalize(local_filepath)})).on('entry', (entry)=>{ //get and unzip to subfolder
                console.log(entry.path); //show every file in archive
              }).on('finish', ()=>{
                console.log('[3.2]: File: '+file+' unzipped');
                console.log('--------------------------------------------------');
                callback();
              })
            })
          } else {
            callback();
          }
        }, function(err, res){
          if(err) console.log(err)
          console.log('\x1b[36m','[!!!] Updates for '+region+'/'+folder+' complete','\x1b[0m');
          console.log('=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=');
          callback();
        })
      })
    }, function(err, res){
      if(err) console.log(err)
      console.log('\x1b[35m','[!!!]: All folders and files for '+region+' downloaded','\x1b[0m');
      console.log('=============================================================');
      callback();
    })
  }, function(err, res){
    if(err) console.log(err);
    console.log('\x1b[35m','===== ALL DONE!!! =====','\x1b[0m');
    c.end();
  })
})
```

But sometimes process freeze and nothing happend (see picture). path.normalizer don't help with it. I try to catch errors with `.on('error')` but nothing. I use MacOS 10.14.4, Node.js v11.10.1.
![Снимок экрана 2019-04-05 в 1 44 08](https://user-images.githubusercontent.com/3946629/55588690-d6d3a580-5747-11e9-978f-594b15f0dc1e.png)

**Comment by MRayermannMSFT** [comment]
@ZJONSSON , I opened a PR to do the normalize fix: #121

**Comment by Shailaputri** [comment]
I am still facing this issue. Some files and folders went missing upon extraction on web hosting platform (00webhost). Has anyone found a work around?
